### DISCUSSION TITLE
Generating Compressed FITS Files (Easily)

### DISCUSSION THREAD

**Discussion by wcleveland** [nom-tam-fits/nom-tam-fits#322] [Q&A]
I couldn't find an example of how to generate a compressed FITS file using the library, and ended up using the following method which seems to work:

```java
GZIPOutputStream gzs = new GZIPOutputStream(new FileOutputStream(filename+".gz"));
BufferedDataOutputStream out = new BufferedDataOutputStream(gzs);
fits.write(out);
gzs.finish(); // I don't think this is necessary.
out.close();
```

Is there an "official" way of doing this? If not would we like to streamline this a little and support other compression file formats?

In the absence of an official method, I can write some documentation on how to do whole file compression using different compression formats that does not increase the dependencies of our library.

Thanks,
Bill

**Comment by ritchieGitHub** [comment]
They are supported ;)
nom.tam.fits.test.CompressTest.testFile()
but only as input stream, this could of course be extended to output streams ;)

**Comment by wcleveland** [comment]
Okay. I'll see what needs to be done to add "external file compression" (CFITSIO terminology) for new files to the library. My technique above is enough to get file compression with the current library, but I'll plan the modifications for my API 2.0 branch.

**Comment by wcleveland** [comment]
Let's push this to 2.0 milestone.
